tools/caffe.cpp

```diff
@@ -63,6 +64,10 @@ class __Registerer_##func { \
 __Registerer_##func g_registerer_##func; \
 }
 
+// Hack to convert macro to string
+#define STRINGIZE(m) #m
+#define STRINGIZE2(m) STRINGIZE(m)
```

Comment:
No need to duplicate these macros... why don't we put them in a common header file (e.g., `common.hpp`)?